packages/api/cms-api/src/generator/generate-crud.ts

```diff
@@ -462,7 +462,16 @@ function generateService({ generatorOptions, metadata }: { generatorOptions: Cru
     const { classNameSingular, fileNameSingular, classNamePlural } = buildNameVariants(metadata);
     const { hasPositionProp, positionGroupProps } = buildOptions(metadata, generatorOptions);
 
-    const positionGroupType = positionGroupProps.length ? `{ ${positionGroupProps.map((prop) => `${prop.name}: ${prop.type}`).join(",")} }` : false;
+    const positionGroupType = positionGroupProps.length
+        ? `{ ${positionGroupProps
+              .map(
+                  (prop) =>
+                      `${prop.name}${prop.nullable ? `?` : ``}: ${
+                          [ReferenceType.EMBEDDED, ReferenceType.SCALAR].includes(prop.reference) ? prop.type : "string"
```

Comment:
and maybe change this check to `[ReferenceType.ManyToOne, ReferenceType.OneToOne].includes(prop.reference) ? "string" : prop.type` to better match the other check.

Reply:
Yes, please change that. Why does this even work with embedded and scalar?

Reply:
I haven't tested embedded or scalar. But I'm assuming forwarding any type should work, because it's just used in an equal-check.